Refs #36537 - Drop description from setting search

girijaasoni opened the pull request:
The description is no longer a field so it can't be searched anymore. This is
not a real fix, but prevents internal server errors on the settings search.

The description field was removed in the following PR: 
Fixes: #de56ffb28064 ("Fixes #34305 - stop creating settings in DB (#9050)")

Community post leading to the decision: 
[https://community.theforeman.org/t/setting-search-returns-error-500/34226](url)

stejskalleos commented:
ok to test

ekohl reviewed (changes requested):
Descriptions of commits matter and this is not a clear description. I'd also argue this is not a fix for https://projects.theforeman.org/issues/36537. In that case we use `Refs`.

I suspect this was introduced in de56ffb2806473db9dc0cfd1f13d38076c7e56ea.

So I'd suggest:

```
Refs #36537 - Drop description from setting search

The description is no longer a field so it can't be searched anymore. This is
not a real fix, but prevents internal server errors on the settings search.

Fixes: de56ffb28064 ("Fixes #34305 - stop creating settings in DB (#9050)")
```

And I wonder if the real fix isn't to add `:description` to the `delegate` statement.

girijaasoni commented:
> Descriptions of commits matter and this is not a clear description. I'd also argue this is not a fix for https://projects.theforeman.org/issues/36537. In that case we use `Refs`.
> 
> I suspect this was introduced in [de56ffb](https://github.com/theforeman/foreman/commit/de56ffb2806473db9dc0cfd1f13d38076c7e56ea).
> 
> So I'd suggest:
> 
> ```
> Refs #36537 - Drop description from setting search
> 
> The description is no longer a field so it can't be searched anymore. This is
> not a real fix, but prevents internal server errors on the settings search.
> 
> Fixes: de56ffb28064 ("Fixes #34305 - stop creating settings in DB (#9050)")
> ```
> 
> And I wonder if the real fix isn't to add `:description` to the `delegate` statement.

Thanks @ekohl , it was a really helpful insight. I wonder if I shall add the community post on this as well :)

stejskalleos commented:
@girijaasoni Code looks good, but what @ekohl is suggesting in the previous comment is changing the commit message itself, not just the description of PR.
I mean it's not bad to have an updated PR description as well, but more important is the commit message, because later when you go over the commits and history of the repo (aka `git log`, `git blame`, ...) you don't see description of the PR, just the commit's description.

girijaasoni commented:
> @girijaasoni Code looks good, but what @ekohl is suggesting in the previous comment is changing the commit message itself, not just the description of PR. I mean it's not bad to have an updated PR description as well, but more important is the commit message, because later when you go over the commits and history of the repo (aka `git log`, `git blame`, ...) you don't see description of the PR, just the commit's description.

Thanks @stejskalleos , updated it :D

stejskalleos commented:
> Thanks @stejskalleos , updated it :D

Are you sure? Looks same to me: https://github.com/theforeman/foreman/pull/9783/commits/6dd4763a1954cb522f5e46ab05801d7067521fca

girijaasoni committed:
Refs #36537 - Drop description from setting search

The description is no longer a field so it can't be searched anymore. This is
not a real fix, but prevents internal server errors on the settings search.

Fixes: de56ffb28064

stejskalleos commented:
[test katello]

stejskalleos reviewed (approved):
Katello job is failing on `test/models/rpm_test.rb:50` ([link](https://github.com/Katello/katello/blob/master/test/models/rpm_test.rb#L47)), but looking at the test I don't think it's related to these changes.

@ekohl any other comments, I'm :green_apple: to merge
